securesystemslib/keys.py

```diff
@@ -1586,6 +1589,40 @@ def is_pem_private(pem, keytype='rsa'):
 
 
 
+def import_ed25519key_from_private_json(json_str, password=None):
+  if password is not None:
+    # This check will not fail, because a mal-formatted passed password fails
+    # above and an entered password will always be a string (see get_password)
+    # However, we include it in case PASSWORD_SCHEMA or get_password changes.
+    securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
+
+    # Decrypt the loaded key file, calling the 'cryptography' library to
+    # generate the derived encryption key from 'password'.  Raise
+    # 'securesystemslib.exceptions.CryptoError' if the decryption fails.
+    key_object = securesystemslib.keys.\
+                 decrypt_key(json_str.decode('utf-8'), password)
+
+  else:
+    logger.debug('No password was given. Attempting to import an'
+        ' unencrypted file.')
+    key_object = json.loads(json_str.decode('utf-8'))
+
+  # Raise an exception if an unexpected key type is imported.
```

Comment:
Do we have an ed25519 schema for this check?
https://github.com/secure-systems-lab/securesystemslib/blob/1c4161e0a8b249e42f4f8d02ffef49a45a2e4006/securesystemslib/formats.py#L308-L315